m365-developer-proxy-plugins/Guidance/GraphSelectGuidancePlugin.cs

```diff
@@ -1,37 +1,146 @@
+    {
+        var modified = false;
+        var versions = new[] { "v1.0", "beta" };
+        foreach (var version in versions)
+        {
+            try
+            {
+                var file = new FileInfo(Path.Combine(proxyFolder, "GraphProxyPlugins", $"graph-{version.Replace(".", "_")}-openapi.yaml"));
+                if (file.LastWriteTime.Date == DateTime.Now.Date)
+                {
+                    // file already updated today
+                    continue;
+                }
+
+                var url = $"https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/openapi/{version}/openapi.yaml";
+                var client = new HttpClient();
+                var response = await client.GetStringAsync(url);
+                File.WriteAllText(file.FullName, response);
+                modified = true;
+            }
+            catch (Exception ex)
+            {
+                _logger?.LogError(ex.Message);
+            }
+        }
+
+        if (modified)
+        {
+            LoadOpenAPIFiles(proxyFolder);
+        }
+    }
+
+    private async void LoadOpenAPIFiles(string proxyFolder)
+    {
+        var versions = new[] { "v1.0", "beta" };
+        foreach (var version in versions)
+        {
+            var file = new FileInfo(Path.Combine(proxyFolder, "GraphProxyPlugins", $"graph-{version.Replace(".", "_")}-openapi.yaml"));
+            if (!file.Exists)
+            {
+                continue;
+            }
+
+            var openApiDocument = await new OpenApiStreamReader().ReadAsync(file.OpenRead());
+            _openApiDocuments[version] = openApiDocument.OpenApiDocument;
+        }
+    }
+
+    private async Task AfterResponse(object? sender, ProxyResponseArgs e)
+    {
         Request request = e.Session.HttpClient.Request;
         if (_urlsToWatch is not null && e.HasRequestUrlMatch(_urlsToWatch) && WarnNoSelect(request))
             _logger?.LogRequest(BuildUseSelectMessage(request), MessageType.Warning, new LoggingContext(e.Session));
     }
 
-    private static bool WarnNoSelect(Request request) =>
-        ProxyUtils.IsGraphRequest(request) &&
-        request.Method == "GET" &&
-        !request.RequestUri.AbsolutePath.EndsWith("/$value", StringComparison.OrdinalIgnoreCase) &&
-        !request.Url.Contains("$select", StringComparison.OrdinalIgnoreCase) &&
-        !request.Url.Contains("%24select", StringComparison.OrdinalIgnoreCase);
+    private bool WarnNoSelect(Request request)
+    {
+        if (!ProxyUtils.IsGraphRequest(request) ||
+            request.Method != "GET")
+        {
+            return false;
+        }
+
+        var graphVersion = ProxyUtils.GetGraphVersion(request.RequestUri.AbsoluteUri);
+        var tokenizedUrl = GetTokenizedUrl(request.RequestUri.AbsoluteUri);
+
+        if (EndpointSupportsSelect(graphVersion, tokenizedUrl))
+        {
+            return !request.Url.Contains("$select", StringComparison.OrdinalIgnoreCase) &&
```

Comment:
There's a missing case here, on beta `select` is also accepted without the `$` prefix